flexmeasures/cli/tests/test_data_add.py

```diff
@@ -87,3 +96,197 @@ def test_cli_help(app):
+                method="add",
+                args=["@sensor_2"],
+            ),
+            dict(method="resample_events", args=["2h"]),
+        ],
+        final_df_output="df_agg",
+    )
+
+    return reporter_config_raw
+
+
+@pytest.mark.skip_github
+def test_add_reporter(app, db, setup_dummy_data, reporter_config_raw):
+    from flexmeasures.cli.data_add import add_report
+
+    sensor1, sensor2, report_sensor = setup_dummy_data
+    report_sensor_id = report_sensor.id
+
+    runner = app.test_cli_runner()
+
+    cli_input_params = {
+        "sensor-id": report_sensor_id,
+        "reporter-config-file": "reporter_config.json",
+        "reporter": "PandasReporter",
+        "start": "2023-04-10T00:00:00 00:00",
+        "end": "2023-04-10T10:00:00 00:00",
+        "output_file": "test.csv",
+    }
+
+    cli_input = to_flags(cli_input_params)
+
+    with runner.isolated_filesystem():
+
+        # save reporter_config to a json file
+        with open("reporter_config.json", "w") as f:
+            json.dump(reporter_config_raw, f)
+
+        # call command
+        result = runner.invoke(add_report, cli_input)
+
+        print(result)
+
+        assert result.exit_code == 0  # run command without errors
+
+        assert "Reporter PandasReporter found" in result.output
+        assert "Report computation done." in result.output
+
+        # Check report is save to the database
+        report_sensor = (
+            db.session.query(Sensor).filter(Sensor.id == report_sensor_id).one_or_none()
+        )  # get fresh report sensor instance
+        stored_report = report_sensor.search_beliefs(
+            event_starts_after=cli_input_params.get("start").replace(" ", "+"),
+            event_ends_before=cli_input_params.get("end").replace(" ", "+"),
+        )
+        assert len(stored_report) == 5
+
+        assert os.path.exists("test.csv")  # check that the file has been created
+        assert (
+            os.path.getsize("test.csv") > 100
+        )  # bytes. Check that the file is not empty
+
+    ##
+    previous_command_end = cli_input_params.get("end").replace(" ", "+")
+
+    cli_input_params = {
+        "sensor-id": report_sensor_id,
+        "reporter-config-file": "reporter_config.json",
+        "reporter": "PandasReporter",
+        "output_file": "test.csv",
+        "timezone": "UTC",
+    }
+
+    cli_input = to_flags(cli_input_params)
+
+    with runner.isolated_filesystem():
+
+        # save reporter_config to a json file
+        with open("reporter_config.json", "w") as f:
+            json.dump(reporter_config_raw, f)
+
+        # call command
+        result = runner.invoke(add_report, cli_input)
+
+        print(result)
+
+        assert result.exit_code == 0  # run command without errors
+
+        assert "Reporter PandasReporter found" in result.output
+        assert "Report computation done." in result.output
+
+        # Check if the report is saved to the database
+        report_sensor = (
```

Comment:
Good point! This is because when we run the command with `runner.invoke` we lose the database session and the object appears as *unbounded*. The same happens in other places, for instance in [test_data_edit.py](https://github.com/FlexMeasures/flexmeasures/blob/6e9c32fc4829777249d1346b1fbceb031cb50f27/flexmeasures/cli/tests/test_data_edit.py#L29).